DevEdu.API/DevEdu.API/Controllers/CourseController.cs

```diff
@@ -28,40 +34,50 @@ public class CourseController : Controller
             _courseRepository = courseRepository;
             _courseService = courseService;
             _mapper = mapper;
+            //_claimsIdentity = this.User.Identity as ClaimsIdentity;
         }
 
         [HttpGet("{id}/simple")]
-        [Description("Get course by id with groups")]
-        [ProducesResponseType(typeof(CourseInfoFullOutputModel), StatusCodes.Status200OK)]
-        public CourseInfoFullOutputModel GetCourseSimple(int id)
+        [Description("Get course by id with topics")]
+        [Authorize]
+        [ProducesResponseType(typeof(CourseInfoShortOutputModel), StatusCodes.Status200OK)]
+
+        public CourseInfoShortOutputModel GetCourseSimple(int id)
         {
             var course = _courseService.GetCourse(id);
-            return _mapper.Map<CourseInfoFullOutputModel>(course);
+            return _mapper.Map<CourseInfoShortOutputModel>(course);
         }
 
         [HttpGet("{id}/full")]
         [Description("Get course by id full")]
+        [AuthorizeRoles(Role.Admin, Role.Manager, Role.Teacher, Role.Methodist)]
```

Comment:
Admin must be excluded from this list
please check the implementation of AuthorizeRolesAttribute and you will find the reason

Reply:
+